app/models/popular_post.rb

```diff
@@ -0,0 +1,43 @@
+class PopularPost < ActiveRecord::Base
+  belongs_to :site
+  belongs_to :post
+
+  scope :published, -> { joins(:post).merge(Post.published) }
+  scope :ordered, -> { order(:rank) }
+
+  class << self
+    def create_data_for_all_site
+      @client = Ranking::ApiClient.new
```

Comment:
Mustn't use instance variable in class level context except it must be.
Because it holds instance variable forever.

It is ok to use class method as facade, but some instance should process.

```ruby
# FIXME builder? importer? or something else...
class PopularPostBuilder
  class << self
    def build(site)
      new(site).build
    end
  end

  def initialize(site)
    @site = site
  end

  def build
    ...
  end

  private

  def client
    @client ||= Client.new
  end
end
```  

------------------------

ここはクラスのコンテキストなので不用意にインスタンス変数使ってはいけません。

`PopularPost.create_data_for_all_site` が
`PopularPost.create_data` の結果に影響を与えてしまいます。

ファサードとしてのクラスメソッドはありだと思いますが、実処理は何かに委ねる形にしてください。